tests/test_views.py

```diff
@@ -0,0 +1,801 @@
+    event2.linking_objects.add(linking_object)
+    event2.save()
+
+    request = rf.get('/')
+    response = views.findEvent(request, linking_object.object_identifier)
+
+    # Determine which event is the oldest.
+    new_event, old_event = sorted([event1, event2], key=lambda e: e.event_date_time)
+
+    # Check that only the oldest event is present in the xml content.
+    assert old_event.event_identifier in response.content
+    assert new_event.event_identifier not in response.content
+
+
+class TestAppAgent:
+    """Tests for views.app_agent."""
+    CONTENT_TYPE = 'application/atom+xml'
+
+    def test_list_returns_ok(self, rf):
+        request = rf.get('/')
+        response = views.app_agent(request)
+        assert response.status_code == 200
+
+    def test_list_response_content_type(self, rf):
+        request = rf.get('/')
+        response = views.app_agent(request)
+        assert response.get('Content-Type') == self.CONTENT_TYPE
+
+    @pytest.mark.xfail(reason='EmptyPage exception is not handled.')
+    def test_list_with_invalid_page(self, rf):
+        request = rf.get('/?page=3')
+        views.app_agent(request)
+
+    @pytest.fixture
+    def app_agent_xml(self):
+        xml = """<?xml version="1.0"?>
+            <premis:agent xmlns:premis="info:lc/xmlns/premis-v2">
+              <premis:agentIdentifier>
+                <premis:agentIdentifierValue>{identifier}</premis:agentIdentifierValue>
+                <premis:agentIdentifierType>PES:Agent</premis:agentIdentifierType>
+              </premis:agentIdentifier>
+              <premis:agentName>asXbNNQgsgbS</premis:agentName>
+              <premis:agentType>Software</premis:agentType>
+            </premis:agent>
+        """
+        identifier = 'EqLVtAeVnHoR'
+        return identifier, xml.format(identifier=identifier)
+
+    def test_post_returns_created(self, rf, app_agent_xml):
+        _, xml = app_agent_xml
+        request = rf.post('/', xml, content_type='application/xml')
+        response = views.app_agent(request)
+        assert response.status_code == 201
+
+    def test_post_response_content_type(self, rf, app_agent_xml):
+        _, xml = app_agent_xml
+        request = rf.post('/', xml, content_type='application/xml')
+        response = views.app_agent(request)
+        assert response.get('Content-Type') == self.CONTENT_TYPE
+
+    def test_post_response_content(self, rf, app_agent_xml):
+        identifier, xml = app_agent_xml
+        request = rf.post('/', xml, content_type='application/xml')
+        response = views.app_agent(request)
+        assert identifier in response.content
+
+    @pytest.mark.xfail(reason='POST request without body raises uncaught exception.')
+    def test_post_without_body_is_handled(self, rf):
+        request = rf.post('/')
+        views.app_agent(request)
+
+    def test_returns_bad_request(self, rf):
+        """Check that a request that is not GET, POST, PUT, or DELETE will
```

Comment:
The intent of the
```
    else:
        return HttpResponseBadRequest("Invalid method for this URL.")
```
seems more about checking to see that if there is not an identifier, you should only be doing a GET or a POST, not a PUT or a DELETE. We may want to open an issue that we should allow HEAD requests. The test to me makes more sense if it is testing that a PUT or DELETE  request without an identifier returns a 400.